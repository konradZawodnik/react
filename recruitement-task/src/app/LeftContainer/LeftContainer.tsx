"use client";
import Image from "next/image";
import { Button } from "@/components/ui/button";
import { Label } from "@/components/ui/label";
import Avatar from "../../../public/Avatar.png";
import {
  LucideBriefcaseMedical,
  LucideCalendarDays,
  LucideChartNoAxesCombined,
  LucideCircleHelp,
  LucideHeadset,
  LucideHospital,
  LucideHouse,
  LucideLayers,
  LucideLogOut,
  LucideNotebookTabs,
  LucideSettings,
} from "lucide-react";

export default function LeftContainer() {
  return (
    <div className="bg-[#fefefe] flex flex-col h-1/2 mt-8 ml-8 pt-8">
      <div className="bg-[#fefefe] flex flex-col ml-4 mr-4 h-40 w-60">
        <Image
          className="dark:invert hover:bg-white mr-2 my-1"
          src={Avatar}
          alt="Avatar"
          width={80}
          height={80}
          priority
        />
        <Label className="font-[700] text-[18px] text-[#1a3f7a]">
          Imię Nazwisko
        </Label>
        <Label className="text-[16px]">Operator</Label>
        <Label className="text-[16px]">name@gmail.com</Label>
      </div>
      <div className="bg-[#d7d8db] border mt-4 w-60"></div>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideHouse className="font-bold text-black" width={24} height={24} />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Strona Główna
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideHeadset
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Wizyty inline
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideBriefcaseMedical
          className="font-bold text-[#0068fa]"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[700] text-[16px] text-[#0068fa]">
          Wizyty domowe
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideHospital
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Wizyty stacjonarne
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideLayers className="font-bold text-black" width={24} height={24} />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Druga opinia
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideNotebookTabs
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Dziennik aktywności
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideCalendarDays
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Kalendarz specjalistów
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideChartNoAxesCombined
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Raporty
        </Label>
      </Button>
      <div className="bg-[#d7d8db] border w-60"></div>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideSettings
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Ustawienia
        </Label>
      </Button>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideCircleHelp
          className="font-bold text-black"
          width={24}
          height={24}
        />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          FAQ
        </Label>
      </Button>
      <div className="bg-[#d7d8db] border w-60"></div>
      <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none">
        <LucideLogOut className="font-bold text-black" width={24} height={24} />
        <Label className="cursor-pointer font-[500] text-[16px] text-[#242628]">
          Wyloguj się
        </Label>
      </Button>
      <div className="bg-[#fefefe] h-2/5 mb-8"></div>
    </div>
  );
}
