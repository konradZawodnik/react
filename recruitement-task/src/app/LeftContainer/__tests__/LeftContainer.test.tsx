import { render } from "@testing-library/react";
import LeftContainer from "../LeftContainer";

it("renders LeftContainer unchanged", () => {
  const { container } = render(<LeftContainer />);
  expect(container).toMatchSnapshot();
});
