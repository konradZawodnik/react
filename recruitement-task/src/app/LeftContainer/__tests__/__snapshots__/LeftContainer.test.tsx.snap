// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LeftContainer unchanged 1`] = `
<div>
  <div
    class="bg-[#fefefe] flex flex-col h-1/2 mt-8 ml-8 pt-8"
  >
    <div
      class="bg-[#fefefe] flex flex-col ml-4 mr-4 h-40 w-60"
    >
      <img
        alt="Avatar"
        class="dark:invert hover:bg-white mr-2 my-1"
        data-nimg="1"
        decoding="async"
        height="80"
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style="color: transparent;"
        width="80"
      />
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[18px] text-[#1a3f7a]"
      >
        Imię Nazwisko
      </label>
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-[16px]"
      >
        Operator
      </label>
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-[16px]"
      >
        name@gmail.com
      </label>
    </div>
    <div
      class="bg-[#d7d8db] border mt-4 w-60"
    />
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-house font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
        />
        <path
          d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Strona Główna
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-headset font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 11h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-5Zm0 0a9 9 0 1 1 18 0m0 0v5a2 2 0 0 1-2 2h-1a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2h3Z"
        />
        <path
          d="M21 16v2a4 4 0 0 1-4 4h-5"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Wizyty inline
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-briefcase-medical font-bold text-[#0068fa]"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 11v4"
        />
        <path
          d="M14 13h-4"
        />
        <path
          d="M16 6V4a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v2"
        />
        <path
          d="M18 6v14"
        />
        <path
          d="M6 6v14"
        />
        <rect
          height="14"
          rx="2"
          width="20"
          x="2"
          y="6"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[700] text-[16px] text-[#0068fa]"
      >
        Wizyty domowe
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-hospital font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6v4"
        />
        <path
          d="M14 14h-4"
        />
        <path
          d="M14 18h-4"
        />
        <path
          d="M14 8h-4"
        />
        <path
          d="M18 12h2a2 2 0 0 1 2 2v6a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2v-9a2 2 0 0 1 2-2h2"
        />
        <path
          d="M18 22V4a2 2 0 0 0-2-2H8a2 2 0 0 0-2 2v18"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Wizyty stacjonarne
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-layers font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.83 2.18a2 2 0 0 0-1.66 0L2.6 6.08a1 1 0 0 0 0 1.83l8.58 3.91a2 2 0 0 0 1.66 0l8.58-3.9a1 1 0 0 0 0-1.83z"
        />
        <path
          d="M2 12a1 1 0 0 0 .58.91l8.6 3.91a2 2 0 0 0 1.65 0l8.58-3.9A1 1 0 0 0 22 12"
        />
        <path
          d="M2 17a1 1 0 0 0 .58.91l8.6 3.91a2 2 0 0 0 1.65 0l8.58-3.9A1 1 0 0 0 22 17"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Druga opinia
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-notebook-tabs font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 6h4"
        />
        <path
          d="M2 10h4"
        />
        <path
          d="M2 14h4"
        />
        <path
          d="M2 18h4"
        />
        <rect
          height="20"
          rx="2"
          width="16"
          x="4"
          y="2"
        />
        <path
          d="M15 2v20"
        />
        <path
          d="M15 7h5"
        />
        <path
          d="M15 12h5"
        />
        <path
          d="M15 17h5"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Dziennik aktywności
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-calendar-days font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2v4"
        />
        <path
          d="M16 2v4"
        />
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="4"
        />
        <path
          d="M3 10h18"
        />
        <path
          d="M8 14h.01"
        />
        <path
          d="M12 14h.01"
        />
        <path
          d="M16 14h.01"
        />
        <path
          d="M8 18h.01"
        />
        <path
          d="M12 18h.01"
        />
        <path
          d="M16 18h.01"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Kalendarz specjalistów
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-chart-no-axes-combined font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 16v5"
        />
        <path
          d="M16 14v7"
        />
        <path
          d="M20 10v11"
        />
        <path
          d="m22 3-8.646 8.646a.5.5 0 0 1-.708 0L9.354 8.354a.5.5 0 0 0-.707 0L2 15"
        />
        <path
          d="M4 18v3"
        />
        <path
          d="M8 14v7"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Raporty
      </label>
    </button>
    <div
      class="bg-[#d7d8db] border w-60"
    />
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-settings font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
        />
        <circle
          cx="12"
          cy="12"
          r="3"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Ustawienia
      </label>
    </button>
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-circle-help font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <path
          d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
        />
        <path
          d="M12 17h.01"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        FAQ
      </label>
    </button>
    <div
      class="bg-[#d7d8db] border w-60"
    />
    <button
      class="items-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none"
    >
      <svg
        class="lucide lucide-log-out font-bold text-black"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
        />
        <polyline
          points="16 17 21 12 16 7"
        />
        <line
          x1="21"
          x2="9"
          y1="12"
          y2="12"
        />
      </svg>
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-[500] text-[16px] text-[#242628]"
      >
        Wyloguj się
      </label>
    </button>
    <div
      class="bg-[#fefefe] h-2/5 mb-8"
    />
  </div>
</div>
`;
